ksqldb-functional-tests/src/test/resources/query-validation-tests/multi-col-keys.json

```diff
@@ -0,0 +1,371 @@
+        "CREATE TABLE OUTPUT AS SELECT K, COUNT(*) as COUNT FROM INPUT WINDOW TUMBLING (SIZE 1 SECOND) GROUP BY K;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": {"K": 1, "K2": 2}, "value": {"V": 0}, "timestamp": 0},
+        {"topic": "input_topic", "key": {"K": 1, "K2": 2}, "value": {"V": 0}, "timestamp": 0},
+        {"topic": "input_topic", "key": {"K": 1, "K2": 2}, "value": {"V": 0}, "timestamp": 1001}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"COUNT": 1}, "window": {"start": 0, "end": 1000, "type": "time"}},
+        {"topic": "OUTPUT", "key": 1, "value": {"COUNT": 2}, "window": {"start": 0, "end": 1000, "type": "time"}},
+        {"topic": "OUTPUT", "key": 1, "value": {"COUNT": 1}, "window": {"start": 1000, "end": 2000, "type": "time"}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "table", "schema": "K INTEGER KEY, COUNT BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "join with repartition on single key",
+      "properties": {
+        "ksql.key.format.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM S1 (K INT KEY, K2 INT KEY, V INT) WITH (kafka_topic='s1', format='JSON');",
+        "CREATE STREAM S2 (K INT KEY, K2 INT KEY, V INT) WITH (kafka_topic='s2', format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM S1 JOIN S2 WITHIN 1 DAY on S1.K = S2.K;"
+      ],
+      "inputs": [
+        {"topic": "s1", "key": {"K": 1, "K2": 2}, "value": {"V": 0}},
+        {"topic": "s2", "key": {"K": 1, "K2": 2}, "value": {"V": 0}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"S1_K2": 2, "S1_V": 0, "S2_K": 1, "S2_K2": 2, "S2_V": 0}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "S1_K INTEGER KEY, S1_K2 INTEGER, S1_V INTEGER, S2_K INTEGER, S2_K2 INTEGER, S2_V INTEGER"}
+        ]
+      }
+    },
+    {
+      "name": "join with repartition on single value",
+      "properties": {
+        "ksql.key.format.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM S1 (K INT KEY, K2 INT KEY, V INT) WITH (kafka_topic='s1', format='JSON');",
+        "CREATE STREAM S2 (K INT KEY, K2 INT KEY, V INT) WITH (kafka_topic='s2', format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM S1 JOIN S2 WITHIN 1 DAY on S1.v = S2.V;"
+      ],
+      "inputs": [
+        {"topic": "s1", "key": {"K": 1, "K2": 2}, "value": {"V": 0}},
+        {"topic": "s2", "key": {"K": 1, "K2": 2}, "value": {"V": 0}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 0, "value": {"S1_K": 1, "S1_K2": 2, "S2_K": 1, "S2_K2": 2, "S2_V": 0}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "S1_V INTEGER KEY, S1_K INTEGER, S1_K2 INTEGER, S2_K INTEGER, S2_K2 INTEGER, S2_V INTEGER"}
+        ]
+      }
+    },
+    {
+      "name": "nested struct key",
```

Comment:
How is this is a nested struct key?

Reply:
physically, it's a struct within a struct. I'll rename this "struct as column in multi-column key"